src/protocols/bdx/tests/TestBdxTransferSession.cpp

```diff
@@ -80,29 +80,44 @@ CHIP_ERROR ReadAndVerifyTLVString(nlTestSuite * inSuite, void * inContext, const
     return err;
 }
 
-// Helper method for verifying that a PacketBufferHandle contains a valid BDX header and message type matches expected.
-void VerifyBdxMessageType(nlTestSuite * inSuite, void * inContext, const System::PacketBufferHandle & msg, MessageType expected)
+CHIP_ERROR AttachHeaderAndSend(TransferSession::MessageTypeData typeData, chip::System::PacketBufferHandle msgBuf,
+                               TransferSession & receiver)
 {
-    CHIP_ERROR err      = CHIP_NO_ERROR;
-    uint16_t headerSize = 0;
-    PayloadHeader payloadHeader;
-
-    if (msg.IsNull())
+    chip::PayloadHeader payloadHeader;
+    if (typeData.ProtocolId == Protocols::BDX::Id.ToFullyQualifiedSpecForm())
     {
-        NL_TEST_ASSERT(inSuite, false);
-        return;
+        payloadHeader.SetMessageType(Protocols::BDX::Id, typeData.MessageType);
```

Comment:
If the typedata stored a Protocols::Id, could just have a single `SetMessageType` call here.

Reply:
Good catch, will implement